### ISSUE TITLE
[gatsby-source-filesystem] unhandled rejection in createRemoteFileNode even if caught

### ISSUE THREAD

**Issue by antoinerousseau** [gatsbyjs/gatsby#14178]
## Description

When calling `createRemoteFileNode` in `gatsby-node`'s `sourceNodes`, if a remote URL fails to download (e.g. if it's a 404), the whole process crashes with an `UNHANDLED REJECTION` error, even if we `.catch`.

https://github.com/antoinerousseau/gatsby-bug-mvce

### Steps to reproduce

- Create a new Gatsby project (`gatsby new gatsby-bug-mvce`)
- Add these lines to `gatsby-node.js`:
```js
const { createRemoteFileNode } = require("gatsby-source-filesystem")

exports.sourceNodes = ({ actions, store, cache, createNodeId }) => {
  const { createNode } = actions

  return createRemoteFileNode({
    url: "https://httpstat.us/404",
    store,
    cache,
    createNode,
    createNodeId,
  })
  .catch(err => {
    console.log("caught error:", err)
  })
  .then(node => {
    console.log("created node:", node)
  })
}
```
https://github.com/antoinerousseau/gatsby-bug-mvce/commits/master

### Actual result

```
yarn run v1.15.2
$ gatsby build
success open and validate gatsby-configs — 0.004 s
success load plugins — 0.239 s
success onPreInit — 0.004 s
success delete html and css files from previous builds — 0.010 s
success initialize cache — 0.015 s
success copy gatsby files — 0.037 s
success onPreBootstrap — 0.006 s
caught error: failed to process https://httpstat.us/404
HTTPError: Response code 404 (Not Found)
error UNHANDLED REJECTION

  Error: failed to process https://httpstat.us/404
  HTTPError: Response code 404 (Not Found)

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

### Expected result

A caught rejection means it's handled, so the process should display:
```
...
caught error: failed to process https://httpstat.us/404
created node: undefined
```
and continue.

### Environment

```
  System:
    OS: macOS 10.14.4
    CPU: (8) x64 Intel(R) Core(TM) i5-8259U CPU @ 2.30GHz
    Shell: 5.3 - /bin/zsh
  Binaries:
    Node: 11.14.0 - /usr/local/bin/node
    Yarn: 1.15.2 - /usr/local/bin/yarn
    npm: 6.9.0 - /usr/local/bin/npm
  Languages:
    Python: 2.7.15 - /usr/local/bin/python
  Browsers:
    Chrome: 74.0.3729.157
    Firefox: 66.0.3
    Safari: 12.1
  npmPackages:
    gatsby: ^2.5.5 => 2.5.5
    gatsby-image: ^2.1.0 => 2.1.0
    gatsby-plugin-manifest: ^2.1.1 => 2.1.1
    gatsby-plugin-offline: ^2.1.1 => 2.1.1
    gatsby-plugin-react-helmet: ^3.0.12 => 3.0.12
    gatsby-plugin-sharp: ^2.0.37 => 2.0.37
    gatsby-source-filesystem: ^2.0.36 => 2.0.36
    gatsby-transformer-sharp: ^2.1.19 => 2.1.19
  npmGlobalPackages:
    gatsby-cli: 2.5.9
```

### Related

https://github.com/gatsbyjs/gatsby/pull/12348#issuecomment-493219565

**Comment by leonfs** [comment]
Have you checked your `package-lock.json` is pointing to the right version?

**Comment by antoinerousseau** [comment]
@leonfs https://github.com/antoinerousseau/gatsby-bug-mvce/blob/master/yarn.lock#L4673

**Comment by wardpeet** [comment]
@antoinerousseau fix is up :)

**Comment by antoinerousseau** [comment]
@wardpeet wow you are way too fast! :D 👍

**Comment by wardpeet** [comment]
Fixed in gatsby-source-filesystem@2.0.37
